### DISCUSSION TITLE
Preventing Drags from Happening in a Particular Droppable Column

### DISCUSSION THREAD

**Discussion by tjv255** [clauderic/dnd-kit#1049] [Q&A] [answered]
In my application, I have several columns. Some columns are to have drag disabled for all items within them, but will still have drop enabled. 

My question is whether I can conditionally block an item from being dragged. 

Here are a couple things I tried:
- Disabling SortableContext
  - Allows me to place items into the column, but does not allow me to move the item's position once placed.
- Throwing an error in onDragStart
  - Works as desired other than it being messy. Nobody wants to see a bunch of error messages in their console.

For just disabling drop is easy as you can cancel drag in the onDragEnd hook, but it doesn't seem like the onDragStart is cancellable (unless you throw an error). Does anyone know if there is any way to get this to work? Any suggestions will be appreciated. Thanks.

**Comment by tjv255** [comment]
So I was playing around I found an answer.

For my draggable, I am using the useSortable hook. From this hook, I get a 'listeners' argument that I use as a property for my item container. To disable drag for the item, you simply don't add the listeners property to it.

const {
    attributes,
    listeners,
    setNodeRef,
    transform,
    transition,
    isDragging
  } = useSortable({ id });
  const newListneners= !dragDisabled && { ...listeners };

  return (
    <div ref={setNodeRef} style={style} {...attributes} {...newListneners}>

I am still open to suggestions if anyone else has any inputs on my method, or if they have another method that they would like to share.

**Comment by lukewatts** [comment, marked as answer]
Just FYI the examples in the docs do this by removing the ref 
`return (<div ref={! isDraggable ? undefined : setNodeRef})`

Could be. a bit cleaner.
